tree/ntuple/v7/src/RPageStorage.cxx

```diff
@@ -686,20 +699,66 @@ void ROOT::Experimental::Internal::RPagePersistentSink::CommitSealedPage(Descrip
 
 std::vector<ROOT::Experimental::RNTupleLocator>
 ROOT::Experimental::Internal::RPagePersistentSink::CommitSealedPageVImpl(
-   std::span<RPageStorage::RSealedPageGroup> ranges)
+   std::span<RPageStorage::RSealedPageGroup> ranges, const std::vector<bool> &mask)
 {
    std::vector<ROOT::Experimental::RNTupleLocator> locators;
+   locators.reserve(mask.size());
```

Comment:
Since we don't know a priori how many bits of the mask are set, we risk committing more memory than needed here. If we expect the mask to potentially have a considerable size, I'd add a `shrink_to_fit()` after the for loops.